src/databricks/labs/ucx/hive_metastore/federation.py

```diff
@@ -14,13 +18,26 @@
 )
 
 from databricks.labs.ucx.account.workspaces import WorkspaceInfo
+from databricks.labs.ucx.assessment.secrets import SecretsMixin
 from databricks.labs.ucx.config import WorkspaceConfig
 from databricks.labs.ucx.hive_metastore import ExternalLocations
 
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class ExtHms:
+    # This is a dataclass that represents the external Hive Metastore connection information
+    db_type: str
+    host: str
+    port: str
+    database: str
+    user: str | None
+    password: str | None
+    version: str | None
```

Comment:
Do we expect more login methods later? It is not uncommon to pack all these together in an authentication string

Reply:
I use it to explode the string because we need them discreet.